ui/desktop/src/contexts/SessionStreamContext.tsx

```diff
@@ -0,0 +1,362 @@
+import React, {
+  createContext,
+  useContext,
+  useRef,
+  useState,
+  useEffect,
+  useCallback,
+  ReactNode,
+} from 'react';
+import { Session } from '../api';
+import { getApiUrl } from '../config';
+
+const TextDecoder = globalThis.TextDecoder;
+
+/**
+ * Event types from the session SSE stream
+ */
+type SessionEvent =
+  | { type: 'Session'; session: Session }
+  | { type: 'Error'; error: string }
+  | { type: 'Ping' };
```

Comment:
this type mimics what we declare on the server. we should use the server one. also this has an extra Ping one.